exchanges/btcmarkets/btcmarkets_wrapper.go

```diff
@@ -272,64 +290,50 @@ func (b *BTCMarkets) FetchOrderbook(p currency.Pair, assetType asset.Item) (orde
 // UpdateOrderbook updates and returns the orderbook for a currency pair
 func (b *BTCMarkets) UpdateOrderbook(p currency.Pair, assetType asset.Item) (orderbook.Base, error) {
 	var orderBook orderbook.Base
-	orderbookNew, err := b.GetOrderbook(p.Base.String(),
-		p.Quote.String())
+	tempResp, err := b.GetOrderbook(b.FormatExchangeCurrency(p, assetType).String(), 2)
 	if err != nil {
 		return orderBook, err
 	}
-
-	for x := range orderbookNew.Bids {
+	for x := range tempResp.Bids {
 		orderBook.Bids = append(orderBook.Bids, orderbook.Item{
-			Amount: orderbookNew.Bids[x][1],
-			Price:  orderbookNew.Bids[x][0],
-		})
+			Amount: tempResp.Bids[x].Volume,
+			Price:  tempResp.Bids[x].Price})
 	}
-
-	for x := range orderbookNew.Asks {
+	for y := range tempResp.Asks {
 		orderBook.Asks = append(orderBook.Asks, orderbook.Item{
-			Amount: orderbookNew.Asks[x][1],
-			Price:  orderbookNew.Asks[x][0],
-		})
-	}
-
-	orderBook.Pair = p
-	orderBook.ExchangeName = b.Name
-	orderBook.AssetType = assetType
-
-	err = orderBook.Process()
-	if err != nil {
-		return orderBook, err
+			Amount: tempResp.Asks[y].Volume,
+			Price:  tempResp.Asks[y].Price})
+		orderBook.Pair = p
```

Comment:
This will process an orderbook for every ask iteration. Please move the below outside the for loop